### PR TITLE
Increase ccache maxsize to 20G

### PR DESCRIPTION
**Pull request by segeljakt**
Addresses #277

### CODE REVIEW

File: arc-mlir/arc-mlir-build [hunk 1]
```diff
@@ -41,6 +41,7 @@ A2M_ASSERTS="${A2M_ASSERTS:+-DLLVM_ENABLE_ASSERTIONS=ON}"
 
 # Do we want ccache?
 A2M_CCACHE="${A2M_CCACHE:+-DLLVM_CCACHE_BUILD=1}"
+A2M_CCACHE_MAXSIZE="${A2M_CCACHE:+-DLLVM_CCACHE_MAXSIZE=20G}"
```
Comment (frej) [inline arc-mlir/arc-mlir-build:44]: I'd prefer if this was only done on the CI-machine, users have their own preferences for the location and size of their ccache setup. If I did this I'd:

1. Figure out a path on the CI-machine which is writeable by the runner, but not zapped by it between runs and call this `<PERSIST_DIR>`

2. Create a shell script:
  ```
 #!/bin/bash
mkdir -p ${PERSIST_DIR}/ccache-cachedir
envsubst > ${PERSIST_DIR}/ccache-config <<EOF
max_size = 20G
cache_dir = ${PERSIST_DIR}/ccache-cachedir
EOF
  ```
3. In the configuration for the workflow environment I would set:
 ```
PERSIST_DIR="<PERSIST_DIR>"
CCACHE_CONFIGPATH="${PERSIST_DIR}/ccache-config"
```
And then configure the workflow to run the shell script before it runs `arc-build`

**Review by frej — CHANGES_REQUESTED** [review]
Ccache configuration should not be dictated by arc-mlir-build, see the inline comment.

**Comment by frej** [comment]
To speed up the build set `BUILD_FLAVOUR=Release` `A2M_ASSERTS="1"` when executing `arc-mlir-build`. That will drastically speed up linking, but still give check assertions. The less understandable backtraces won't be a problem as we are not debugging on the CI machine.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -21,8 +21,14 @@ jobs:
     - name: Update submodule
       run: git submodule update --init --recursive
 
+    - name: Persist artifacts
+      run: ./.github/persist.sh
+      env:
+        PERSIST_DIR: /root/actions-runner/persist/
+        CCACHE_CONFIGPATH=/root/actions-runner/persist/ccache-config
```
Comment (frej) [inline .github/workflows/test.yml:28]: Check if there is a global environment for all actions, this way `CCACHE_CONFIGPATH` will have to be repeated for the "Build and test arc-mlir and arc-script" action.

The syntax looks fishy: Why is it `:` in one assignment and `=` in the other.

Comment (segeljakt) [reply .github/workflows/test.yml:28]: This explains why the syntax highlighting didnt work 😅

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -21,8 +21,14 @@ jobs:
...
+    - name: Persist artifacts
+      run: ./.github/persist.sh
+      env:
+        PERSIST_DIR: /root/actions-runner/persist/
+        CCACHE_CONFIGPATH: /root/actions-runner/persist/ccache-config
+
       # Arc-MLIR specific tests
-    - name: Build and test arc-mlir
+    - name: Build and test arc-mlir and arc-script
       run: |
```
Comment (frej) [inline .github/workflows/test.yml:32]: I suspect `CCACHE_CONFIGPATH` won't be set for this action.

**Review by frej — APPROVED** [review]
If this works I'm happy with the implementation.

File: .github/persist.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+mkdir -p ${PERSIST_DIR}/ccache-cachedir
```
Comment (frej) [inline .github/persist.sh:2]: A late addition, wrap the existing body in:
```
if [[ -d "${PERSIST_DIR}/ccache-cachedir" ]] then
    echo "The Ccache directory exists at ${PERSIST_DIR}/ccache-cachedir"
else
   echo "Creating Ccache directory at ${PERSIST_DIR}/ccache-cachedir"
  ... old stuff ...
fi
```
That way we can check the logs to verify that the directory isn't removed by the runner

**Pull request closed without merging**